features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScope.kt

```diff
@@ -149,25 +169,33 @@ internal class RumSessionScope(
 
         if (isInteraction || isApplicationStartEvent) {
             if (isNewSession || isExpired || isTimedOut) {
-                renewSession(nanoTime)
+                val reason = if (isNewSession) {
+                    StartReason.USER_APP_LAUNCH
+                } else if (isExpired) {
+                    StartReason.INACTIVITY_TIMEOUT
+                } else {
+                    StartReason.MAX_DURATION
+                }
+                renewSession(nanoTime, reason)
             }
             lastUserInteractionNs.set(nanoTime)
         } else if (isExpired) {
             if (backgroundTrackingEnabled && isBackgroundEvent) {
-                renewSession(nanoTime)
+                renewSession(nanoTime, StartReason.INACTIVITY_TIMEOUT)
```

Comment:
I think here we can qualify it as `background_launch`, maybe? If I'm not wrong, we can hit this line in case if application/session was started by some background event (but I may have outdated memories of this logic).

Reply:
As per the [ticket](https://datadoghq.atlassian.net/browse/RUM-1336): 

> [!Note]
> The following will not be implemented yet because we don’t track non-interactive sessions.